### PR TITLE
Transform

### PR DESCRIPTION
**Pull request by decel337**

### CODE REVIEW

File: RenderEngine/RenderEngine/Basic/Vector3.cs [hunk 1]
```diff
@@ -1,4 +1,9 @@
-﻿namespace RenderEngine.Basic;
+﻿using RenderEngine.Transformer;
+using System.Runtime.CompilerServices;
+using System.Security.Cryptography.X509Certificates;
```
Comment (Akara4ok) [inline RenderEngine/RenderEngine/Basic/Vector3.cs:3]: Clean up usings

File: RenderEngine/RenderEngine/Shapes/Sphere.cs [hunk 2]
```diff
@@ -1,10 +1,21 @@
 ﻿using RenderEngine.Basic;
 using RenderEngine.Interfaces;
+using RenderEngine.Transformer;
 
 namespace RenderEngine.Shapes;
 
-public record Sphere(Vector3 Orig, float Radius) : IShape
+public record Sphere : IShape
```
Comment (Akara4ok) [inline RenderEngine/RenderEngine/Shapes/Sphere.cs:7]: change to class

File: RenderEngine/RenderEngine/Core/Scene.cs [hunk 3]
```diff
@@ -13,4 +14,18 @@ public Scene(IReadOnlyList<IShape> shapes, IReadOnlyList<ILighting> lighting)
         Shapes = shapes;
         Lighting = lighting;
     }
+
+    public void Transform(Transform transform)
```
Comment (st1lson) [inline RenderEngine/RenderEngine/Core/Scene.cs:18]: The same I guess

File: RenderEngine/RenderEngine/Transformer/Matrix.cs [hunk 4]
```diff
@@ -0,0 +1,23 @@
+﻿namespace RenderEngine.Transformer
+{
+    public class Matrix
+    {
+        public static float[,] Multiply(float[,] matrix1, float[,] matrix2)
+        {
+            float[,] multiplyMatrix = new float[matrix1.GetLength(0), matrix2.GetLength(1)];
+
+            for (int row = 0; row < matrix1.GetLength(0); row++)
```
Comment (st1lson) [inline RenderEngine/RenderEngine/Transformer/Matrix.cs:9]: So the class is useless to be non-static

File: RenderEngine/RenderEngine/Transformer/Transform.cs [hunk 5]
```diff
@@ -0,0 +1,102 @@
+﻿using RenderEngine.Basic;
+
+namespace RenderEngine.Transformer
+{
+    public class Transform
+    {
+        public static float[,] IdentityMatrix { get; } = new float[,]
+            {
+                { 1, 0, 0, 0 },
```
Comment (st1lson) [inline RenderEngine/RenderEngine/Transformer/Transform.cs:9]: readonly?

**Review by st1lson — CHANGES_REQUESTED** [review]

File: RenderEngine/RenderEngine/Transformer/Matrix.cs (hunk 4 above)
Comment (decel337) [reply RenderEngine/RenderEngine/Transformer/Matrix.cs:9]: not understand

Comment (st1lson) [reply RenderEngine/RenderEngine/Transformer/Matrix.cs:9]: The class is non-static and contains only one static method, so the class can be marked as a static

**Review by st1lson — APPROVED** [review]

**Pull request merged**